apis/mcp/v1alpha1/controlplane_types.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2023 Upbound Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ControlPlaneParameters are the configurable fields of a ControlPlane.
+type ControlPlaneParameters struct {
+	// Description is the description of the the control plane
+	Description string `json:"description,omitempty"`
+
+	// Configuration is the name of the predefined configuration
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Configuration *string `json:"configuration"`
```

Comment:
```suggestion
	Configuration string `json:"configuration"`
```

Required fields should not be pointer.